### DISCUSSION TITLE
Pimcore X: Error UnableToSetVisibility

### DISCUSSION THREAD

**Discussion by choppe3766** [pimcore/pimcore#10438] [Q&A / Support]
Hi all,

i've installed pimcore X with docker like described in the skeleton GitHub repo.
Everything works fine during the installation, no errors.

Then in admin the login screen is shown, but after a successful login i get this error:

An exception has been thrown during the rendering of a template ("Unable to set visibility for file minified_javascript_core_5ef804a750161f5ed3e446db13d78749.js. chmod(): Operation not permitted").

All directories in var/* has the www-data as owner and all directories are writeable.
Does somebody has the same problem and how can i solve it?

Thanks for your help.
Chris

**Comment by AlexanderWeckmer** [comment]
Me too, Did you solve it?

**Reply by BjBue** [reply]
Hello, I got the same Problem!
Did anybody maybe has an idea what the problem could be?
The dokumentation is saying nothing about, i have tried reinstalling the whole instance. Same Error again.
Dear PimcoreTeam, why is there no answer? Is this a id10t-Error? Help would be nice.

**Reply by kingjia90** [reply]
@BjBue Hi, it looks like something out of the scope of Pimcore but more related to the configuration of one's own server/system (permissions, user/groups, security measures of shared hostings or so), since `chmod(): Operation not permitted` may comes directly from the local system. 
Make sure this step was properly applied https://github.com/pimcore/skeleton/blob/b408879dd362a441646d0d1f7eb20c208c629d06/README.md?plain=1#L36-L37 when using Docker, or `chown` everything to `www-data`.

**Reply by BjBue** [reply]
Thank you so much for reply. Yes, i have done this steps. I am working on docker and yes, you are damn right.
After checking the directory in the docker container everything belongs to root!
Many,many thank you for your help!
I will report further actions.

**Reply by BjBue** [reply]
Well than:
I have chown all files to www-data, cleared the cache (symfony and pimcore) but nothing changed.
Same error as before:

`An exception has been thrown during the rendering of a template ("Unable to set visibility for file minified_javascript_core_869125b43f96f60d106ca1db9b697985.js. chmod(): Operation not permitted").`

**Reply by kingjia90** [reply]
What about `var` folder permisssions? user group of www-data? did you run recursively(with option `-R`), right?

**Reply by BjBue** [reply]
Sir, yes, Sir! ;-)
Sorry I am kidding.
I have done an chown -R www-data:www-data html/ and because now changings i have testet an chmod -R 777 html/
and no changes. always the same error (see above).
I found the files in var/admin , all belong to www-data and all have 777 permissions.
I cleared the cache (symfony/pimcore), i restarted and rebuild the docker container, nothing changed.
I have absolutly no more ideas

**Reply by BjBue** [reply]
... and, by the way, many, many thanks for your help!

**Reply by BjBue** [reply]
O.K. it was the simplest solution and i should have do it from the begining.
Shoe on my head!
I searched the files" minified_javascript_core*" and found them in ./var/admin. Than i delete them.
Tadaaaa! That solve the problem!
Maybee anybody can move this solution to the top? It cost me 2 days now and make me headache! It would be great when everybody who comes in this situation find this solution faster than me!
Thank everybody for the help!

**Comment by choppe3766** [comment]
Hi,

I assume you work in a docker container.
Do you has set the correct user in the docker file like described in this git repo under topic 3?
https://github.com/pimcore/skeleton/ <https://github.com/pimcore/skeleton/>

This solves the same problem in my Pimcore instance ;-)

Cheers Chris

**Comment by choppe3766** [comment]
I have only changed the „user“ lines in the docker file and restarted the docker container and everything works.
Maybe if you change the directory and file permissions on your host machine the docker container have a problem with the permissions?
Additionally you can change the permissions to 777 recursively for the var directory… but this is not a secure way ;-)

**Comment by choppe3766** [comment]
By the way, the files are located in var/admin, maybe you can make a „chmod -R 777 var/admin“
